### PR TITLE
refactor(sentry): add per request session tracking

### PR DESCRIPTION
**Pull request by subotic**

### CODE REVIEW

File: shttps/Server.cpp [hunk 1]
```diff
@@ -721,6 +774,11 @@ namespace shttps {
     }
     //=========================================================================
 
+    // process request is one thread that is waiting for incoming requests
+    // and processes them. It is started by the main thread and will run
+    // until the server is stopped.
+    // This is in addition to the thread pool that is used to process the
+    // lua scripts.
```
Comment (BalduinLandolt) [inline shttps/Server.cpp:781]: does this mean that all non-lua logic is single threaded?

**Review by BalduinLandolt — APPROVED** [review]

File: shttps/Server.cpp (hunk 1 above)
Comment (subotic) [reply shttps/Server.cpp:781]: no, everything is multithreaded and this comment is a bit misleading. There is only one thread pool used by all routes. I will correct it.

Comment (subotic) [reply shttps/Server.cpp]: @BalduinLandolt can you maybe take again a look at the changes?

Comment (subotic) [reply shttps/Server.cpp]: > does this mean that all non-lua logic is single threaded?

also please note the new naming of the methods. Before we had `process_request` and `processRequest`. I hope the new naming is a bit more clearer what is happening (at least what I think that is happening).

File: shttps/Server.cpp [hunk 2]
```diff
@@ -721,7 +724,9 @@ namespace shttps {
     }
     //=========================================================================
 
-    static void *process_request(void *arg) {
+    // The socket_request_processor is added to each tread and initiates handling of requests
```
Comment (BalduinLandolt) [inline shttps/Server.cpp:727]:
```suggestion
    // The socket_request_processor is added to each thread and initiates handling of requests
```

**Review by BalduinLandolt — APPROVED** [review]
looks good/better, just found one typo

File: shttps/Server.cpp (hunk 2 above)
Comment (subotic) [reply shttps/Server.cpp]: thanks, changed now.

**Pull request merged**